lib/improver/tests/nowcasting/lightning/test_NowcastLightning.py

```diff
@@ -0,0 +1,946 @@
+                set_up_cube(num_realization_points=3)))
+        threshold_coord = self.precip_cube.coord('realization')
+        threshold_coord.points = [0.5, 7.0, 35.0]
+        threshold_coord.rename('threshold')
+        threshold_coord.units = cf_units.Unit('mm hr-1')
+        self.precip_cube.rename("probability_of_precipitation")
+        self.precip_cube.attributes.update({'relative_to_threshold': 'above'})
+        self.precip_cube.data[1:, 0, ...] = 0.
+
+    def test_basic(self):
+        """Test that the method returns the expected cube type"""
+        plugin = Plugin()
+        result = plugin.apply_precip(self.fg_cube, self.precip_cube)
+        self.assertIsInstance(result, Cube)
+
+    def test_input(self):
+        """Test that the method does not modify the input cubes."""
+        plugin = Plugin()
+        cube_a = self.fg_cube.copy()
+        cube_b = self.precip_cube.copy()
+        plugin.apply_precip(cube_a, cube_b)
+        self.assertArrayAlmostEqual(cube_a.data, self.fg_cube.data)
+        self.assertArrayAlmostEqual(cube_b.data, self.precip_cube.data)
+
+    def test_nearby_threshold_low(self):
+        """Test that the method accepts a threshold point within machine
+        tolerance."""
+        self.precip_cube.coord('threshold').points = [0.5000000001, 7., 35.]
+        plugin = Plugin()
+        plugin.apply_precip(self.fg_cube, self.precip_cube)
+
+    def test_missing_threshold_low(self):
+        """Test that the method raises an error if the precip_cube doesn't
+        have a threshold coordinate for 0.5."""
+        self.precip_cube.coord('threshold').points = [1.0, 7., 35.]
+        plugin = Plugin()
+        msg = ("No matching any precip cube for")
+        with self.assertRaisesRegex(ConstraintMismatchError, msg):
+            plugin.apply_precip(self.fg_cube, self.precip_cube)
+
+    def test_missing_threshold_mid(self):
+        """Test that the method raises an error if the precip_cube doesn't
+        have a threshold coordinate for 7.0."""
+        self.precip_cube.coord('threshold').points = [0.5, 8., 35.]
+        plugin = Plugin()
+        msg = ("No matching high precip cube for")
+        with self.assertRaisesRegex(ConstraintMismatchError, msg):
+            plugin.apply_precip(self.fg_cube, self.precip_cube)
+
+    def test_missing_threshold_high(self):
+        """Test that the method raises an error if the precip_cube doesn't
+        have a threshold coordinate for 35.0."""
+        self.precip_cube.coord('threshold').points = [0.5, 7., 20.]
+        plugin = Plugin()
+        msg = ("No matching intense precip cube for")
+        with self.assertRaisesRegex(ConstraintMismatchError, msg):
+            plugin.apply_precip(self.fg_cube, self.precip_cube)
+
+    def test_precip_zero(self):
+        """Test that zero precip probs reduce lightning risk"""
+        plugin = Plugin()
+        expected = set_up_cube_with_no_realizations()
```

Comment:
Could we have initialisation as `expected = self.fg_cube.copy()` throughout please - it makes the tests more robust and easier to understand.

Reply:
I agree.